File: ruby/SWFdump.rb
require File.dirname(__FILE__)+'/SWF.rb'

if ARGV.length != 1 
  print "Usage: SWFtest.rb <swf file>\n"
  exit 1;
end
data = IO.read(ARGV[0])
data.force_encoding("BINARY") if RUBY_VERSION >= '1.9'
swf = SWF.new(data);
swf.dump()

File: perl/swfdump.pl
#! /usr/bin/perl

require 'SWF.pm';

use strict;
use warnings;
use LWP::Simple;
use SWF;


if (@ARGV != 1) {
    print "Usage: swfdump.pl <swffile>\n";
    exit (1);
}

my ($swffile) = @ARGV;
my $swfdata = get('file://'.$swffile);

my $swf = new SWF();
$swf->input($swfdata);

$swf->dump();





